MediaBrowser.Controller/Persistence/IItemRepository.cs

```diff
@@ -17,6 +17,13 @@ namespace MediaBrowser.Controller.Persistence
     /// </summary>
     public interface IItemRepository : IDisposable
     {
+        /// <summary>
+        /// Gets a database connection.
+        /// </summary>
+        /// <param name="readOnly">A value indicating whether the connection is readonly.</param>
+        /// <returns>The connection.</returns>
+        ManagedConnection GetConnection(bool readOnly = false);
```

Comment:
Note that this would give plugins direct access to the database

Reply:
How could we remedy this?

Reply:
I think this extraction has the same effect
At the moment I'm not sure how to fix this but I don't believe direct db access is something we want plugins to have..

Reply:
Yeah I reverted it since it breaks too much anyway and does not provide soemthing.

But we still need a way to access the DB within the migrations. The only difference between now and before is that we provide a direct interface for accessing the DB. It's not like plugins were unable to do it before, just required more manual work.